File: frontend/src/lib/api/organizationContext.ts
/**
 * API service for organization context settings
 */

import axiosInstance from '../axios';

export interface OrganizationContextSettings {
  context_text: string;
  include_in_default_prompts: boolean;
  include_in_custom_prompts: boolean;
  is_shared: boolean;
  using_shared_from: string | null;
}

export interface OrganizationContextUpdate {
  context_text?: string;
  include_in_default_prompts?: boolean;
  include_in_custom_prompts?: boolean;
  is_shared?: boolean;
}

export interface SharedOrganizationContext {
  user_id: string;
  owner_name: string;
  owner_role: string;
  context_text: string;
  is_active: boolean;
}

export interface SharedOrganizationContextList {
  shared_contexts: SharedOrganizationContext[];
}

export async function getOrganizationContext(): Promise<OrganizationContextSettings> {
  const response = await axiosInstance.get('/user-settings/organization-context');
  return response.data;
}

export async function updateOrganizationContext(
  settings: OrganizationContextUpdate
): Promise<OrganizationContextSettings> {
  const response = await axiosInstance.put('/user-settings/organization-context', settings);
  return response.data;
}

export async function resetOrganizationContext(): Promise<{
  message: string;
  default_settings: OrganizationContextSettings;
}> {
  const response = await axiosInstance.delete('/user-settings/organization-context');
  return response.data;
}

export async function getSharedOrganizationContexts(): Promise<SharedOrganizationContextList> {
  const response = await axiosInstance.get('/user-settings/organization-context/shared');
  return response.data;
}

export async function useSharedOrganizationContext(
  userId: string | null
): Promise<OrganizationContextSettings> {
  const response = await axiosInstance.post('/user-settings/organization-context/use-shared', {
    user_id: userId,
  });
  return response.data;
}

File: backend/app/schemas/organization_context.py
"""
Pydantic schemas for organization context settings.

These schemas define the request/response models for user-level organization
context that is injected into LLM system prompts during summarization.
"""

from typing import Optional

from pydantic import BaseModel
from pydantic import Field

from app.core.constants import DEFAULT_ORG_CONTEXT_INCLUDE_CUSTOM_PROMPTS
from app.core.constants import DEFAULT_ORG_CONTEXT_INCLUDE_DEFAULT_PROMPTS
from app.core.constants import DEFAULT_ORG_CONTEXT_TEXT
from app.core.constants import ORG_CONTEXT_MAX_LENGTH


class OrganizationContextSettings(BaseModel):
    """
    Response schema for user organization context settings.

    Contains the organization context text and toggle preferences.
    """

    context_text: str = Field(
        default=DEFAULT_ORG_CONTEXT_TEXT,
        max_length=ORG_CONTEXT_MAX_LENGTH,
        description="Organization/project background context for LLM summaries",
    )
    include_in_default_prompts: bool = Field(
        default=DEFAULT_ORG_CONTEXT_INCLUDE_DEFAULT_PROMPTS,
        description="Include context when using system default summary prompts",
    )
    include_in_custom_prompts: bool = Field(
        default=DEFAULT_ORG_CONTEXT_INCLUDE_CUSTOM_PROMPTS,
        description="Include context when using user-created custom prompts",
    )
    is_shared: bool = Field(
        default=False,
        description="Whether this org context is shared with all users",
    )
    using_shared_from: Optional[str] = Field(
        default=None,
        description="User ID whose shared context this user is using (null = own)",
    )

    model_config = {
        "json_schema_extra": {
            "examples": [
                {
                    "context_text": "Greenleaf Analytics is a healthcare data company building the Pulse platform...",
                    "include_in_default_prompts": True,
                    "include_in_custom_prompts": False,
                    "is_shared": False,
                    "using_shared_from": None,
                }
            ]
        }
    }


class OrganizationContextUpdate(BaseModel):
    """
    Request schema for updating organization context settings.

    All fields are optional - only provided fields will be updated.
    """

    context_text: Optional[str] = Field(
        default=None,
        max_length=ORG_CONTEXT_MAX_LENGTH,
        description="Organization/project background context for LLM summaries",
    )
    include_in_default_prompts: Optional[bool] = Field(
        default=None,
        description="Include context when using system default summary prompts",
    )
    include_in_custom_prompts: Optional[bool] = Field(
        default=None,
        description="Include context when using user-created custom prompts",
    )
    is_shared: Optional[bool] = Field(
        default=None,
        description="Whether to share this org context with all users",
    )

    model_config = {
        "json_schema_extra": {
            "examples": [
                {
                    "context_text": "Updated organization context...",
                    "include_in_default_prompts": True,
                    "include_in_custom_prompts": True,
                    "is_shared": True,
                }
            ]
        }
    }


class SharedOrganizationContext(BaseModel):
    """A shared organization context from another user."""

    user_id: str
    owner_name: str
    owner_role: str
    context_text: str
    is_active: bool = Field(
        False, description="Whether the current user is using this shared context"
    )


class SharedOrganizationContextList(BaseModel):
    """List of shared org contexts from other users."""

    shared_contexts: list[SharedOrganizationContext] = []
